hummingbot/connector/exchange/huobi/huobi_exchange.py

```diff
@@ -0,0 +1,423 @@
+
+    async def _user_stream_event_listener(self):
+        """
+        This functions runs in background continuously processing the events received from the exchange by the user
+        stream data source. It keeps reading events from the queue until the task is interrupted.
+        The events received are balance updates, order updates and trade events.
+        """
+        async for stream_message in self._iter_user_event_queue():
+            try:
+                channel = stream_message["ch"]
+                if not channel.startswith("accounts") and not channel.startswith("orders") and not channel.startswith("trade.clearing"):
+                    continue
+                data = stream_message["data"]
+                if len(data) == 0 and stream_message["code"] == 200:
+                    # This is a subcribtion confirmation.
+                    self.logger().info(f"Successfully subscribed to {channel}")
+                    continue
+
+                if channel.startswith("accounts"):
+                    asset_name = data["currency"].upper()
+                    balance = data["balance"]
+                    available_balance = data["available"]
+
+                    self._account_balances.update({asset_name: Decimal(balance)})
+                    self._account_available_balances.update({asset_name: Decimal(available_balance)})
+                elif channel.startswith("orders"):
+                    safe_ensure_future(self._process_order_update(data))
+                elif channel.startswith("trade.clearing"):
+                    if data["eventType"] == "trade":
+                        safe_ensure_future(self._process_trade_event(data))
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().error("Unexpected error in user stream listener loop.", exc_info=True)
+                await self._sleep(5.0)
+
+    async def _process_order_update(self, msg: Dict[str, Any]):
+        client_order_id = msg["clientOrderId"]
+        order_status = msg["orderStatus"]
+
+        if order_status not in ["submitted", "partial-filled", "filled", "partially-canceled", "canceled", "canceling"]:
+            self.logger().debug(f"Unrecognized order update response - {msg}")
+
+        tracked_order = self.in_flight_orders.get(client_order_id, None)
+
+        if tracked_order is not None:
+            order_update = OrderUpdate(
+                trading_pair=tracked_order.trading_pair,
+                update_timestamp=self.current_timestamp * 1e-3,
+                new_state=CONSTANTS.ORDER_STATE[order_status],
+                client_order_id=client_order_id
+            )
+            self._order_tracker.process_order_update(order_update=order_update)
+
+    async def _process_trade_event(self, trade_event: Dict[str, Any]):
+        client_order_id = trade_event["clientOrderId"]
+        tracked_order = self.in_flight_orders.get(client_order_id, None)
```

Comment:
To process order trades we have to consider live orders and recently finished orders
```suggestion
        tracked_order = self._order_tracker.fetch_order(client_order_id=client_order_id)
```